### DISCUSSION TITLE
Getting subdomains to work

### DISCUSSION THREAD

**Discussion by maslaral** [coder/code-server#3111] [Q&A]
I've set up my subdomains to try to do some React development. However, I'm having issues where the 3000.mydomain.com just redirects back to my editor (i.e. mydomain.com). I'm using Caddy and have the settings that are in the setup guide, and I've edited my code-server.service file with the _--proxy-domain mydomain.com_ command set. 

Any suggestions for getting this to work properly? I'm using Google domains and I've also got my subdomain set up there (see screenshot below).

![tempsnip](https://user-images.githubusercontent.com/47020246/114432139-f9de8e80-9b85-11eb-9cfb-18da4d8981c8.png)

**Comment by jsjoeio** [comment]
Hey @maslaral - have you looked read the docs related for [Proxying a Create React App](https://github.com/cdr/code-server/blob/main/docs/FAQ.md#proxying-to-create-react-app) by chance?

**Reply by maslaral** [reply]
Yeah, I've read through that and was able to get it to work. But had issues with routing in React. I thought it was possible (and recommended) to use subdomains instead.

**Comment by code-asher** [comment]
Based on the screenshot it looks like it's redirecting `3000.domain.com` to `domain.com`. I think what you want instead is one of these:

1. An A record pointing `3000.domain.com` to your IP (the same thing you probably did for `domain.com`).
2. A CNAME pointing `3000.domain.com` to `domain.com` so it resolves to the same IP.

**Reply by code-asher** [reply]
Actually I'm not sure you can CNAME to the root domain so you'd need the A record.

What I do is create an A record for `cname.domain.com` then a CNAME for `*.domain.com` that points to `cname.domain.com` that way all subdomains resolve to the same IP without having to set up an A record for each port.

**Reply by code-asher** [reply]
I'm not sure why I didn't just create a wildcard A record though hahaha. I'll probably do that instead.

**Reply by maslaral** [reply]
Thanks for the additional info. I set it up with the wildcard A record, getting an SSL error but guessing that's another issue I'll have to try to resolve. Thanks!

**Reply by code-asher** [reply]
No problem! Yeah, you'll need a wildcard certificate or you'll need to set up separate certs for each subdomain.

**Reply by maslaral** [reply]
I'm guessing wildcard certificates are handled via Caddy (at least in my case). Is there an example caddyfile I can look over?

**Reply by code-asher** [reply]
I actually haven't set up wildcard certificates using Caddy yet since they don't support my registrar so currently I just brute force it like a madman:

```
domain.com {
  reverse_proxy localhost:8080
}
3000.domain.com {
  reverse_proxy localhost:8080
}
3001.domain.com {
  reverse_proxy localhost:8080
}
``` 

And so on for each sub-domain I want.

Wildcard certificates through Caddy (via Let's Encrypt) require a DNS challenge (https://caddyserver.com/docs/automatic-https#wildcard-certificates) and I don't think they have a provider for Google domains yet (https://github.com/caddy-dns).

**Reply by maslaral** [reply]
Hmm, I'm still having issues with accessing the subdomain. I'm getting an ERR_NAME_NOT_RESOLVED error, even though I've exactly matched your setup. I'm wondering if adding the --proxy-domain flag in the .service file is working correctly.

**Reply by code-asher** [reply]
I think that means the browser isn't able to resolve the name to an IP address. It could mean the changes to DNS haven't propagated to you yet or that something is misconfigured.

You might try `dig` or `ping` against the sub-domain and see if you get the right IP address.
